src/main/java/com/cs3365/rpgdapi/weapons/WeaponsServiceImpl.java

```diff
@@ -38,4 +38,14 @@ public WeaponEntity findWeapon(UUID identity) throws WeaponsException
             throw new WeaponsException(e.getMessage(), e.getCause());
         }
     }
+
+    @Override
+    public void updateWeapon(WeaponEntity weapon) throws WeaponsException {
+        try {
+            weaponsRepository.save(weapon);
+        } catch(Exception e) {
+            throw new WeaponsException(e.getMessage(), e.getCause());
+        }
+    }
```

Comment:
This method needs to contain the logic present in the `WeaponsController` that I pointed out. Please refactor this to manage it.